### PR TITLE
types btrfs: fix format without swap or subvolumes

### PR DESCRIPTION
**Pull request by kira-bruneau**
Hi! Recently when trying to format a btrfs drive I ran into ```syntax error near unexpected token `fi'```.

It looks like #568 introduced a change that generates an empty if statement if neither the swap config or subvolumes are set and this just adds a guard to avoid adding that empty if block if neither are set.

### CODE REVIEW

File: lib/types/btrfs.nix [hunk 1]
```diff
@@ -120,29 +120,31 @@ in
...
-              SUBVOL_ABS_PATH="$MNTPOINT/${subvol.name}"
-              mkdir -p "$(dirname "$SUBVOL_ABS_PATH")"
-              if ! btrfs subvolume show "$SUBVOL_ABS_PATH" > /dev/null 2>&1; then
-                btrfs subvolume create "$SUBVOL_ABS_PATH" ${toString subvol.extraArgs}
-              fi
-              ${swapCreate "$SUBVOL_ABS_PATH" subvol.swap}
-            )
-          '') (lib.attrValues config.subvolumes)}
-        fi
+        ${lib.optionalString (config.swap != {} && config.subvolumes != []) ''
```
Comment (Lassulus) [inline lib/types/btrfs.nix:123]:
```suggestion
        ${lib.optionalString (config.swap != {} && config.subvolumes != {}) ''
```

Comment (Lassulus) [reply lib/types/btrfs.nix:123]: subvolume is an attrset

Comment (kira-bruneau) [reply lib/types/btrfs.nix:123]: Oops right thanks!

**Comment by Lassulus** [comment]
it would also be nice to have an example config and a test which triggers that error. so we can detect regressions in the future :)

**Comment by Lassulus** [comment]
@mergify queue

**Pull request merged**